src/core_plugins/tile_map/public/kibana_map.js

```diff
@@ -335,6 +337,20 @@ export class KibanaMap extends EventEmitter {
     return this._leafletMap.getBounds();
   }
 
+  getMetersPerPixel() {
+    const pointC = this._leafletMap.latLngToContainerPoint(this._leafletMap.getCenter()); // center (pixels)
+    const pointX = [pointC.x + 1, pointC.y]; // add one pixel to x
+    const pointY = [pointC.x, pointC.y + 1]; // add one pixel to y
+
+    const latLngC = this._leafletMap.containerPointToLatLng(pointC);
+    const latLngX = this._leafletMap.containerPointToLatLng(pointX);
+    const latLngY = this._leafletMap.containerPointToLatLng(pointY);
+
+    const distanceX = latLngC.distanceTo(latLngX); // calculate distance between c and x (latitude)
+    const distanceY = latLngC.distanceTo(latLngY); // calculate distance between c and y (longitude)
```

Comment:
good idea to measure the diagonal. heatmap-grids change aspect-ratio per zoom level, this is a good approximation.